### ISSUE TITLE
Intermittent Slow DNS Resolution

### ISSUE THREAD

**Issue by aleemb** [docker/docker#23927]
I have two identical servers on AWS both running identical docker containers. One of the servers recently started having issues with DNS resolution. Restarting the container fixed the issue but the slow DNS resolution problem returned after ~15 minutes.

```
~> time nslookup example.com
Server:		127.0.0.11
Address:	127.0.0.11#53

Non-authoritative answer:
Name:	example.com
Address: 93.184.216.34


real	0m5.857s
user	0m0.004s
sys	0m0.000s

~> cat /etc/resolv.conf 
search eu-west-1.compute.internal
nameserver 127.0.0.11
options ndots:0
```

The `resolv.conf` settings are the same on the second server's container which is working just fine. This problem comes up every couple of weeks, I am not sure why. The last two times this happened, I redeployed the servers and it went away but three times is definitely a pattern.

**Comment by aleemb** [comment]
```
~$ docker info
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 5
Server Version: 1.11.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 143
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.19.0-59-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: XXXX
Total Memory: XXXX MiB
Name: XXXX
ID: FPNI:OOR3:XNAC:4JNV:NVNQ:4HRY:JRFW:7FXJ:L36Q:L3JL:ZPVW:RNOA
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Username: XXXXX
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Comment by aleemb** [comment]
More details,

The issue is because of the nameserver `127.0.0.11` which as I understand goes back to the host. Changing the name server to the same one on host's resolve.conf (`172.30.0.2`) fixes and reverting back to `127.0.0.11` brings the slow DNS resolution issue back.

**Comment by aleemb** [comment]
Also, if I spin up another instance of the same container on the same server, things work fine in there. It's just that this one container went into some bad state and is very slow in resolve DNS now.

**Comment by aleemb** [comment]
```
~> docker inspect scribe
[
    {
        "Id": "e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33",
        "Created": "2016-06-24T08:15:23.5040748Z",
        "Path": "/bin/bash",
        "Args": [
            "/start.sh"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 24502,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2016-06-24T08:15:23.854321788Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "sha256:cf679e520d0cb69f406d84e4c72f9fa46b0318c41b8439149ca953d1c13e9b19",
        "ResolvConfPath": "/var/lib/docker/containers/e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33/hostname",
        "HostsPath": "/var/lib/docker/containers/e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33/hosts",
        "LogPath": "/var/lib/docker/containers/e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33/e943924cc45e7b5f07a4a2563a0d8539ac6ce4ed7303da3a2b69779d1123be33-json.log",
        "Name": "/scribe",
        "RestartCount": 0,
        "Driver": "aufs",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": [
            "76715a10326b11305a67a55aca8fd606278bd77870466e1d6ae6ecc898b0404c",
            "9b205b6cf6f0546d6e0b5e05493d194076f3a88a2cfdee54a3946d0ada78e038",
            "7330e53117f852397de54d4632b1c845960efe90b22e881ebad8a736dc9ef00b"
        ],
        "HostConfig": {
            "Binds": [],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "scribe_default",
            "PortBindings": {
                "443/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "443"
                    }
                ],
                "80/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "80"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": [],
            "CapAdd": [
                "SYS_PTRACE"
            ],
            "CapDrop": null,
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "StorageOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": -1,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "BlkioIOps": 0,
            "BlkioBps": 0,
            "SandboxSize": 0
        },
        "GraphDriver": {
            "Name": "aufs",
            "Data": null
        },
        "Mounts": [],
        "Config": {
            "Hostname": "e943924cc45e",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "443/tcp": {},
                "80/tcp": {}
            },
            "Tty": true,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "APPID",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NGINX_VERSION=1.10.0",
                "NGINX_LOG_DIR=/var/log/nginx",
                "PHALCON_VERSION=2.0.10",
                "SUNDOWN_VERSION=0.3.11",
                "PHP_REDIS_VERSION=2.2.7",
                "NGINX_REDIS_VERSION=0.3.8",
                "LUA_VERSION=0.10.2",
                "BUILD_PKGS=build-essential software-properties-common language-pack-en-base",
                "RUNTIME_PKGS=nano curl wget git",
                "BUILD_PKGS_NGINX=libpcre3-dev libssl-dev",
                "RUNTIME_PKGS_NGINX=libpcre3 libluajit-5.1-dev",
                "BUILD_PKGS_PHP=php5-dev",
                "RUNTIME_PKGS_PHP=imagemagick memcached php5 php5-curl php5-fpm php5-gd php5-imagick php5-mcrypt php5-memcached php5-mysqlnd phpunit language-pack-ur-base",
                "RUNTIME_PKGS_REDIS=redis-server",
                "RUNTIME_PKGS_SUPERVISOR=supervisor",
                "DEVBOX=false"
            ],
            "Cmd": [
                "/bin/bash",
                "/start.sh"
            ],
            "Image": "aleemb/scribe",
            "Volumes": null,
            "WorkingDir": "/root",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "ac03de205b9c5bccef4b48fdc52a20b3d6db52eaa9e86a68fc3537dbc4b137ce",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "scribe",
                "com.docker.compose.service": "scribe",
                "com.docker.compose.version": "1.6.2"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "a60640c20b333259ff7ea2ac257bc47dff5f61bfcd367b3aac79c6917ed65f47",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "443/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "443"
                    }
                ],
                "80/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "80"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/a60640c20b33",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "scribe_default": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "e943924cc45e",
                        "scribe"
                    ],
                    "NetworkID": "0e81b6975a4f20f3385964220948d1583e266a581712d9aef355c8952197505c",
                    "EndpointID": "9c893f31b06a033d108ae271adf62ca52d60e4500799e58cc598a015176a2667",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:12:00:02"
                }
            }
        }
    }
]
```

**Comment by aleemb** [comment]
Syslog

```
[Fri Jun 24 09:49:25 2016] device veth002b21d entered promiscuous mode
[Fri Jun 24 09:49:25 2016] IPv6: ADDRCONF(NETDEV_UP): veth002b21d: link is not ready
[Fri Jun 24 09:49:25 2016] eth0: renamed from vetha404c1e
[Fri Jun 24 09:49:25 2016] IPv6: ADDRCONF(NETDEV_CHANGE): veth002b21d: link becomes ready
[Fri Jun 24 09:49:25 2016] br-0e81b6975a4f: port 2(veth002b21d) entered forwarding state
[Fri Jun 24 09:49:25 2016] br-0e81b6975a4f: port 2(veth002b21d) entered forwarding state
[Fri Jun 24 09:49:40 2016] br-0e81b6975a4f: port 2(veth002b21d) entered forwarding state
[Fri Jun 24 09:52:40 2016] vetha404c1e: renamed from eth0
[Fri Jun 24 09:52:40 2016] br-0e81b6975a4f: port 2(veth002b21d) entered disabled state
[Fri Jun 24 09:52:40 2016] br-0e81b6975a4f: port 2(veth002b21d) entered disabled state
[Fri Jun 24 09:52:40 2016] device veth002b21d left promiscuous mode
[Fri Jun 24 09:52:40 2016] br-0e81b6975a4f: port 2(veth002b21d) entered disabled state
[Fri Jun 24 09:54:19 2016] aufs au_opts_verify:1612:docker[2182]: dirperm1 breaks the protection by the permission bits on the lower branch
[Fri Jun 24 09:54:19 2016] aufs au_opts_verify:1612:docker[2182]: dirperm1 breaks the protection by the permission bits on the lower branch
[Fri Jun 24 09:54:19 2016] aufs au_opts_verify:1612:docker[2182]: dirperm1 breaks the protection by the permission bits on the lower branch
[Fri Jun 24 09:54:19 2016] aufs au_opts_verify:1612:docker[2182]: dirperm1 breaks the protection by the permission bits on the lower branch
[Fri Jun 24 09:54:19 2016] aufs au_opts_verify:1612:docker[2182]: dirperm1 breaks the protection by the permission bits on the lower branch
[Fri Jun 24 09:54:19 2016] device vethd58f376 entered promiscuous mode
[Fri Jun 24 09:54:19 2016] IPv6: ADDRCONF(NETDEV_UP): vethd58f376: link is not ready
[Fri Jun 24 09:54:19 2016] eth0: renamed from veth3508ee4
[Fri Jun 24 09:54:19 2016] IPv6: ADDRCONF(NETDEV_CHANGE): vethd58f376: link becomes ready
[Fri Jun 24 09:54:19 2016] br-0e81b6975a4f: port 2(vethd58f376) entered forwarding state
[Fri Jun 24 09:54:19 2016] br-0e81b6975a4f: port 2(vethd58f376) entered forwarding state
[Fri Jun 24 09:54:34 2016] br-0e81b6975a4f: port 2(vethd58f376) entered forwarding state
```

**Comment by thaJeztah** [comment]
This may be a duplicate of https://github.com/docker/docker/issues/24604, could you have a look if that looks similar?

/cc @sanimej

**Comment by lennartjuette** [comment]
@thaJeztah : This problem sounds familiar. I'm the author of #24604 .

**Comment by aleemb** [comment]
@thaJeztah Seems to be the same issue. DNS resolution just stops working for no reason, it's bugging the hell out of me :/

**Comment by aleemb** [comment]
Though I should add my setup is on EC2 not KVM or similar VM.

**Comment by thaJeztah** [comment]
@aleemb thanks; let me close this issue, let's continue in #24604

**Comment by sanimej** [comment]
@aleemb @lennartjuette This is a duplicate of https://github.com/docker/docker/issues/22185. It has been fixed in 1.12. And 1.12.1 has been released. Please give it a try.
